include/SH3/glcallback.hpp

```diff
@@ -0,0 +1,21 @@
+/** @file
+ *
+ *  OpenGL debug callback function type definition header.
+ *
+ *  @copyright 2016-2018 Palm Studios
+ *
+ *  @date 13-2-2018
+ *
+ *  @author Jesse Buhagiar
+ */
+#ifndef SH3_GL_CALLBACK_HPP_INCLUDED
+#define SH3_GL_CALLBACK_HPP_INCLUDED
+
+// We use C linkage on this function typedef so we don't get any C++ name decoration (as OpenGL is a C library)
+extern "C"
+{
+    typedef void __glCallback;
```

Comment:
This is not a function though, you're just defining `__glCallback` as `void`?

Reply:
Oops. I was under the assumption the compiler would automatically insert `extern "C"` along with the typedef function.

Reply:
This is not a function typedef though. You're just typedeffing `void`, which is not a function, but a "nothing"-type.
If you wanted a typedef for a `debugCallback`-like function you could do `extern "C" typedef void(*__glCallback)(GLenum source, GLenum type, ...);` or (C++11) `using __glCallback = void(*)(GLenum Source, GLenum type, ...)`. There's also such a thing as function references in C++.
Also note that identifiers starting with a double-underscore are reserved.

Reply:
> If you wanted a typedef for a debugCallback-like function you could do extern "C" typedef void(*__glCallback)(GLenum source, GLenum type, ...); or (C++11) using __glCallback = void(*)(GLenum Source, GLenum type, ...).

Ohhh okay that makes a lot more sense now. It's been a bit of a while since I've dealt with function pointers...

> Also note that identifiers starting with a double-underscore are reserved.

By the compiler??